rfc/text/advanced/ap_27_pubsub_event_history.md

```diff
@@ -1,58 +1,132 @@
 ## Event History {#pubsub-event-history}
 
-Instead of complex QoS for message delivery, a *Broker* may provide *message history*. A *Subscriber* is responsible to handle overlaps (duplicates) when it wants "exactly-once" message processing across restarts.
-
-The *Broker* may allow for configuration on a per-topic basis.
+Instead of complex QoS for message delivery, a *Broker* may provide *message history*. A *Subscriber* is 
+responsible to handle overlaps (duplicates) when it wants "exactly-once" message processing across restarts.
 
 The event history may be transient or persistent message history (surviving *Broker* restarts).
 
-A *Broker* that implements *event history* must (also) announce role `HELLO.roles.callee`, indicate `HELLO.roles.broker.history == 1` and provide the following (builtin) procedures.
+The *Broker* implementation may allow for configuration on a per-topic basis.
+
+To understand what actually event history means lets review the publication flow. When one peer decides to publish
+a message to a topic it results in `Publication` WAMP message with concrete `publication_id`, `payload` and `options`
+among other attributes. The *Broker* in his turn checks current subscriptions and create an `Event` WAMP message for
+every subscription. There can be no subscriptions, one exactly matched to publication topic subscription and zero or 
+more pattern-based subscriptions. Every unique subscription (based on topic URI and options) has unique ID. All
+subscribers of the same subscription get the same subscription ID.
+
+{align="left"}
+    TODO: Some scheme for pub/sub
+
+Event History means the events published to concrete subscription in historical order. Let's see an example.
```

Comment:
"historical order (that is by broker receive time)"

Reply:
👍🏻  added